src/pandas_profiling/compare_reports.py

```diff
@@ -0,0 +1,144 @@
+        # If two keys have an equal value, we take that value as new value
+        # If the values are not equal, we recursively merge them
+
+        return {
+            **d1,
+            **d2,
+            **{
+                k: [d1[k], d2[k]]
+                if _isequal(d1[k], d2[k])
+                else _update_merge(d1[k], d2[k])
+                for k in {*d1} & {*d2}
+            },
+        }
+    else:
+        # This case happens when values are merged
+        # It bundle values in a list, making sure
+        # to flatten them if they are already lists
+
+        if isinstance(d1, list) and isinstance(d2, list):
+            # This is the tuple for alerts
+            return d1, d2
+        elif isinstance(d1, tuple) and isinstance(d2, list):
+            return (*d1, d2)
+        else:
+            return [
+                *(d1 if isinstance(d1, list) else [d1]),
+                *(d2 if isinstance(d2, list) else [d2]),
+            ]
+
+
+def _placeholders(*reports) -> List[ProfileReport]:
+    """Needed for merging interactions"""
+    keys = {v for r in reports for v in r["scatter"].keys()}
+    type_keys = {v for r in reports for v in r["table"]["types"].keys()}
+    for idx in range(len(reports)):
+        # Interactions
+        for k1 in keys:
+            for k2 in keys:
+                if k1 not in reports[idx]["scatter"]:
+                    reports[idx]["scatter"][k1] = {}
+                if k2 not in reports[idx]["scatter"][k1]:
+                    reports[idx]["scatter"][k1][k2] = ""
+
+        # Types
+        for type_key in type_keys:
+            if type_key not in reports[idx]["table"]["types"]:
+                reports[idx]["table"]["types"][type_key] = 0
+
+    return list(reports)
+
+
+def _compare_title(titles: List[str]) -> str:
+    if all(titles[0] == title for title in titles[1:]):
+        return titles[0]
+    else:
+        title = ", ".join(titles[:-1])
+        return f"<em>Comparing</em> {title} <em>and</em> {titles[-1]}"
+
+
+def compare(
+    reports: Sequence[Union[ProfileReport, Dict]],
+    config: Optional[Settings] = None,
+) -> ProfileReport:
+    """
+    Compare Profile reports
+
+    Args:
+        reports: two reports to compare
+                 input may either be a ProfileReport, or the summary obtained from report.get_description()
+        config: the settings object for the merged ProfileReport
+
+    """
+    if len(reports) < 2:
+        raise ValueError("At least two reports are required for this comparison")
+
+    if config is None:
+        config = Settings()
+
+    if len(reports) > 2:
+        warnings.warn(
+            "Comparison of more than two reports is in beta. "
+            "Reports will be produced, but may yield unexpected formatting."
+        )
+
+    if any(not isinstance(report, dict) for report in reports):
+        # Use titles as labels
+        labels = [report.config.title for report in reports]  # type: ignore
```

Comment:
The `type: ignore`s are partially needed because mypy does not pick up on the type narrowing. In 3.10 there is the explicit `TypeGuard`. Having the explicit schema for dataset descriptions will also help (as part of the refactor mentioned a couple of times here now). Refactored part of the code to reduce the 'type: ignore's.